modules/core/arrow-extras/src/main/kotlin/arrow/data/SequenceK.kt

```diff
@@ -41,6 +41,15 @@ data class SequenceK<out A>(val sequence: Sequence<A>) : SequenceKOf<A>, Sequenc
       }
     }
 
+  override fun equals(other: Any?): Boolean =
```

Comment:
@pakoito  @raulraja  what do you guys think about overwriting `equals` in `SequenceK`? I am not quite happy with it and thinking about reverting it. Additional opinions would be great.

Reply:
I don't disagree, it should be a transparent wrapper but AFAIK data class should already do that, right?

Reply:
Yes you are right, but this data class contains a sequence property and sequences do not support `equals` ([source](https://youtrack.jetbrains.com/issue/KT-8511)).

Reply:
Sequence doesn't have an identity, and is potentially infinite, so I understand why it wouldn't work. Why dod you need it for anyway?

Reply:
Exactly! That's why started thinking about it again and created this CR. I needed it for testing the laws but I will revert it and pass an Eq to the law.